### PR TITLE
add in-container vendor

### PR DESCRIPTION
**Pull request by SamuelStuchly**
Add option to run make vendor in container like other targets, to avoid issues with go module caching depending on what you already have on your machine.

### CODE REVIEW

File: hack/go-mod.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+go mod tidy
+go mod vendor
+go mod verify
```
Comment (JoelSpeed) [inline hack/go-mod.sh:4]: Nit: It's good practice to leave an empty new line at the end of files

**Review by JoelSpeed — COMMENTED** [review]
One nit, otherwise fine by me
/approve

**Comment by JoelSpeed** [comment]
@SamuelStuchly please add a PR description to explain the motivation behind this change before we merge
/retest

**Review by elmiko — APPROVED** [review]
/lgtm

**Pull request merged**